src/common/browser-adapters/background-browser-event-manager.ts

```diff
@@ -153,22 +152,17 @@ export class BackgroundBrowserEventManager implements BrowserEventManager {
             })}]`;
             return await this.eventResponseFactory.applyEventTimeout(result, timeoutErrorContext);
         } else {
-            if (result === undefined) {
-                // It is possible that this is the result of a fire and forget listener
-                // wrap promise resolution in 2-minute timeout to ensure it completes during service worker lifetime
-                return await this.eventResponseFactory.applyFireAndForgetDelay(result);
-            } else {
+            if (result !== undefined) {
                 // This indicates a bug in an ApplicationListener; they should always either
                 // return a Promise (to indicate that they are responsible for understanding
-                // how long their response takes to process) or void/undefined (to indicate
-                // that they are "fire and forget", and should trigger the above delay case).
+                // how long their response takes to process) or void/undefined.
```

Comment:
nit: can we include in this comment that void/ undefined means that the listener's work was all sync and has already completed.